matrix_client/client.py

```diff
@@ -379,3 +379,27 @@ def get_user(self, user_id):
         """
 
         return User(self.api, user_id)
+
+    def get_public_rooms_ids(self):
```

Comment:
I understand, I will take my time and figure out how to approach this again.
Since other patches in this series are not dependent on this one, I'll be happy if you could merge them (assuming you are ok with them).